crates/parser/src/grammar/expressions/atom.rs

```diff
@@ -563,7 +563,9 @@ fn closure_expr(p: &mut Parser<'_>) -> CompletedMarker {
     let m = p.start();
 
```

Comment:
We should add a test for this
```suggestion

    // test closure_binder
    // fn main() { for<'a> || (); }
```
and then regenerate the tests with `cargo codegen`

Reply:
Done, for easy review of the updated test output: https://github.com/rust-lang/rust-analyzer/pull/19897/commits/55328ba9b45a6e32e4af281271b5e80fd5ae5418?w=1